src/herbivoresAndCarnivores.js

```diff
@@ -1,19 +1,36 @@
 'use strict';
 
-class Animal {
-  // write your code here
-}
+/**
+ * @param {Object} state
+ * @param {Object[]} actions
+ *
+ * @return {Object[]}
+ */
+function transformStateWithClones(state, actions) {
+  // write code here
+  let stateCopy = { ...state };
+  const actionsList = [];
 
-class Herbivore extends Animal {
-  // write your code here
-}
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        break;
+    }
+    actionsList.push({ ...stateCopy });
+  }
 
-class Carnivore extends Animal {
-  // write your code here
+  return actionsList;
 }
 
-module.exports = {
-  Animal,
-  Herbivore,
-  Carnivore,
-};
+module.exports = transformStateWithClones;
```

Comment:
The provided code does not match the task description. The task requires the implementation of `Animal`, `Herbivore`, and `Carnivore` classes with specific properties and methods. The current function `transformStateWithClones` is unrelated to the task at hand. Please provide the relevant code that pertains to the creation of the specified classes and their functionalities.